src/modules/fw_pos_control/figure_eight/FigureEight.cpp

```diff
@@ -131,15 +102,68 @@ void FigureEight::updateSetpoint(const matrix::Vector2f &curr_pos_local, const m
 	valid_parameters.loiter_radius = math::max(valid_parameters.loiter_radius,
 					 MINIMAL_FEASIBLE_MAJOR_TO_MINOR_AXIS_RATIO * valid_parameters.loiter_minor_radius);
 
-	// Calculate the figure eight pattern points.
-	FigureEightPatternPoints pattern_points;
-	calculateFigureEightPoints(pattern_points, valid_parameters);
+	return valid_parameters;
+}
 
-	// Check if we need to switch to next segment
-	updateSegment(curr_pos_local, valid_parameters,  pattern_points);
+void FigureEight::initializePattern(const matrix::Vector2f &curr_pos_local, const matrix::Vector2f &ground_speed,
+				    const FigureEightPatternParameters &parameters, FigureEightPatternPoints pattern_points)
+{
+	// Initialize the currently active segment, if it hasn't been active yet, or the pattern has been changed.
+	if ((_current_segment == FigureEightSegment::SEGMENT_UNDEFINED) || (_active_parameters != parameters)) {
+		Vector2f rel_pos_to_center;
```

Comment:
@KonradRudin The naming confused me a bit. I thought this is the relative vector from the current position to the center. However, it's the other way around if I'm not mistaken. Can we maybe name it something like `center_to_vehicle_relative` or something

Reply:
I renamed it